libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -495,7 +495,7 @@ ASTPointer<UserDefinedTypeName> ASTJsonImporter::createUserDefinedTypeName(Json:
 	string nameString = member(_node, "name").asString();
 	boost::algorithm::split(strs, nameString, boost::is_any_of("."));
 	for (string s: strs)
-		namePath.push_back(ASTString(s));
+		namePath.emplace_back(s);
```

Comment:
Shouldn't this here use move or at least `const&` above?

Reply:
Haha, what's the point of the whole loop in general :-)?

Reply:
It injects shared_ptrs

Reply:
Or does it?

Reply:
Nope. Does nothing at all :-).

Reply:
Maybe we should actually get rid of ``using ASTString = std::string;`` altogether...

Reply:
We should. The idea behind ASTString was that it could be replaced by a string view at some point, but I guess this is not really beneficial.

Reply:
Did we just keep this idle loop in the end now :-)?

Reply:
Just a style refactoring.